Lib/test/test_frame.py

```diff
@@ -442,5 +461,28 @@ def dummy():
         # The following line should not cause a segmentation fault.
         self.assertIsNone(frame.f_back)
 
+class LocalsTest(unittest.TestCase):
+    """
+    Tests for locals.
+    """
+
+    def test_locals_cur_frame_issue113939(self):
```

Comment:
Ideally the name of a test should be more descriptive (and the issue in a comment):

```suggestion
    def test_locals_cleared_after_exception_handled(self):
        # see gh-113939
```